controllers/orbit.controller.js

```diff
@@ -39,26 +61,29 @@ orbitController.createOrbit = async function(ai) {
 
 	// find possible candidates
     Ai.find({ _id: { $ne: ai._id }}, (err, ais) => {
-		let matchedAis = 0;
-
-		// check for potential matches
-		ais.forEach( (_ai) => {
-			if (isMatch(ai, _ai)) {
-				matchedAis++;
-				// push matched ais to orbit
-				_ai.orbits.forEach( (_orbit) => {
-					orbit.planets.push(_orbit);
-				});
-			}
-		});
+		// only look for a match if other ais are found
+		if (ais.length > 0) {
+			let matchedAis = 0;
+
+			// check for potential matches
+			ais.forEach( (_ai) => {
+				if (isMatch(ai, _ai)) {
+					matchedAis++;
+					// push matched ais to orbit
+					_ai.orbits.forEach( (_orbit) => {
+						orbit.planets.push(_orbit);
+					});
+				}
+			});
 
-		// if no match is found just add the last 3.
-		if (matchedAis == 0) {
-			for (let i = (ais.length - 1); i >= (ais.length - 3); i--) {
-				try {
-					orbit.planets.push(ais[i].orbits[ais[i].orbits.length - 1]);
-				} catch (err) {
-					console.log(err);
+			// if no match is found just add the last 3.
```

Comment:
Pakt hij nu 3 random orbits?

Reply:
This is some older code, but no, not random. Just the last 3. Having no matches would result in an empty orbit. To resolve this it checks whether it has found a match and if it doesn't it will add the last 3 orbits which are the highest in the chain.